src/Illuminate/Queue/Worker.php

```diff
@@ -93,17 +100,20 @@ class Worker
      * @param  \Illuminate\Contracts\Events\Dispatcher  $events
      * @param  \Illuminate\Contracts\Debug\ExceptionHandler  $exceptions
      * @param  callable  $isDownForMaintenance
+     * @param  callable  $setQueueDaemonFlag
      * @return void
      */
     public function __construct(QueueManager $manager,
                                 Dispatcher $events,
                                 ExceptionHandler $exceptions,
-                                callable $isDownForMaintenance)
+                                callable $isDownForMaintenance,
+                                callable $setQueueDaemonFlag)
```

Comment:
This is unfortunately a breaking change. Apps or libraries implementing this class will see their implementation fail be they're not passing the new required dependency.

Reply:
Could this then be merged to 9.x?

Reply:
Can't you make it nullable?

Reply:
Yes, I don't know why I haven't figured that out myself.